osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/endpoints/SmartMeteringConfigurationEndpoint.java

```
@@ -879,4 +883,52 @@ public SetClockConfigurationResponse getSetClockConfigurationResponse(
+            @ScheduleTime final String scheduleTime, @ResponseUrl final String responseUrl) throws OsgpException {
+
+        final GetConfigurationObjectAsyncResponse response = new GetConfigurationObjectAsyncResponse();
+
+        final com.alliander.osgp.domain.core.valueobjects.smartmetering.GetConfigurationObjectRequest dataRequest = this.configurationMapper
+                .map(request,
+                        com.alliander.osgp.domain.core.valueobjects.smartmetering.GetConfigurationObjectRequest.class);
+
+        final String correlationUid = this.configurationService.enqueueGetConfigurationObjectRequest(
+                organisationIdentification, dataRequest.getDeviceIdentification(), dataRequest,
+                MessagePriorityEnum.getMessagePriority(messagePriority),
+                this.configurationMapper.map(scheduleTime, Long.class));
+
+        response.setCorrelationUid(correlationUid);
+        response.setDeviceIdentification(request.getDeviceIdentification());
+
+        this.saveResponseUrlIfNeeded(correlationUid, responseUrl);
+
+        return response;
+    }
+
+    @PayloadRoot(localPart = "GetConfigurationObjectAsyncRequest", namespace = SMARTMETER_CONFIGURATION_NAMESPACE)
+    @ResponsePayload
+    public GetConfigurationObjectResponse getGetConfigurationObjectResponse(
+            @RequestPayload final GetConfigurationObjectAsyncRequest request) throws OsgpException {
+
+        GetConfigurationObjectResponse response = null;
+        try {
+            response = new GetConfigurationObjectResponse();
+            final MeterResponseData meterResponseData = this.meterResponseDataService
+                    .dequeue(request.getCorrelationUid());
+
+            response.setResult(OsgpResultType.fromValue(meterResponseData.getResultType().getValue()));
+            if (meterResponseData.getMessageData() instanceof String) {
```

I think we should throw an error if casting is not ok. Else the description will be empty and further no notification is send to the user. Except that an exception is logged.
<code>
try{
                    response.setDescription((String) meterResponseData.getMessageData());
                }
                catch(final ClassCastException e){
                    throw new OsgpException(ComponentType.DOMAIN_SMART_METERING, "cast exception " + e.getMessage());
                }
</code>